builder/azure/common/template/template_builder.go

```diff
@@ -513,6 +513,7 @@ func (s *TemplateBuilder) SetSecurityProfile(secureBootEnabled bool, vtpmEnabled
 	resource.Properties.SecurityProfile.SecurityType = compute.SecurityTypesTrustedLaunch
 	resource.Properties.SecurityProfile.UefiSettings.SecureBootEnabled = to.BoolPtr(secureBootEnabled)
 	resource.Properties.SecurityProfile.UefiSettings.VTpmEnabled = to.BoolPtr(vtpmEnabled)
+	resource.Properties.SecurityProfile.EncryptionAtHost = to.BoolPtr(encryptionAtHost)
```

Comment:
Can we add an if statement here to check if secureBootEnabled or vtpmEnabled is set, and then set the SecurityType/Uefi Settings, otherwise just set the EncryptionAtHost, like this

```
resource.Properties.SecurityProfile = &compute.SecurityProfile{}
if secureBootEnabled || vtpmEnabled {
	resource.Properties.SecurityProfile.UefiSettings = &compute.UefiSettings{}
	resource.Properties.SecurityProfile.SecurityType = compute.SecurityTypesTrustedLaunch
	resource.Properties.SecurityProfile.UefiSettings.SecureBootEnabled = to.BoolPtr(secureBootEnabled)
	resource.Properties.SecurityProfile.UefiSettings.VTpmEnabled = to.BoolPtr(vtpmEnabled)
	resource.Properties.SecurityProfile.EncryptionAtHost = to.BoolPtr(encryptionAtHost)
}
resource.Properties.SecurityProfile.EncryptionAtHost = to.BoolPtr(encryptionAtHost)

That way we aren't setting trusted launch unless the user enables it, I believe its valid to set EncryptionAtHost but not trusted launch.  When building a VHD with UbuntuServer 16.04-LTS as an example and this was failing due to being incompatible with Trusted Launch, but you can still set encrypt it at host

Reply:
Sure, I will do this but we do not need the double assignment:
```
	resource.Properties.SecurityProfile.EncryptionAtHost = to.BoolPtr(encryptionAtHost)
}
resource.Properties.SecurityProfile.EncryptionAtHost = to.BoolPtr(encryptionAtHost)

```

I will try to cover all tomorrow.

Reply:
Oops yes, great catch!